src/signalhandler.cpp

```diff
@@ -59,7 +60,7 @@ SignalHandler::~SignalHandler() {
 
 void SignalHandler::pipeReadReady() {
   int signal;
-  if (read(m_pipefds[0], &signal, sizeof(signal)) == sizeof(signal)) {
+  while (read(m_pipefds[0], &signal, sizeof(signal)) == sizeof(signal)) {
     logger.log() << "Signal" << signal;
     emit quitRequested();
```

Comment:
Im not sure about emitting multiple quitRequested signal. Maybe we should emit just one of them and ignore the rest.

Reply:
If we get multiple POSIX signals queued up, we always emit multiple `quitRequested` signals, it just takes an extra trip through the Qt even loop to wake up the `QSocketNotifier` again if there is still data waiting in the pipe to be read. However, the `while` loop doesn't seem to make a difference to this bug, so I can put back to the way it was.